tests/models/test_sparse.py

```diff
@@ -0,0 +1,73 @@
+"""Compare the outputs of a sparse model running sparse vs sparse model running dense.
+Note: sparse kernels do not have bitwise correctness vs the dense models. 
+As a result, in this test, we just confirm that the top selected tokens of the 
+sparse models are in the top N selections of same model running dense.
+Run `pytest tests/models/test_sparse.py --forked`.
+"""
+
+import gc
+import pytest
+import torch
+from compare_utils import check_logprobs_close
+
+model_format_pairs = [
+    ("nm-testing/Llama-2-7b-pruned50-retrained", "sparse_w16a16"),
+    ("nm-testing/TinyLlama-1.1B-Chat-v1.0-pruned2.4",
+     "semi_structured_sparse_w16a16"),
+    ("nm-testing/OpenHermes-2.5-Mistral-7B-pruned50", "sparse_w16a16"),
+    ("nm-testing/OpenHermes-2.5-Mistral-7B-pruned2.4",
+     "semi_structured_sparse_w16a16"),
+]
+
+
+@pytest.mark.parametrize("model_format_pairs", model_format_pairs)
+@pytest.mark.parametrize("dtype", ["half"])
+@pytest.mark.parametrize("max_tokens", [32])
+@pytest.mark.parametrize("num_logprobs", [3])
+def test_models(
+    vllm_runner_sparse,
+    example_prompts,
+    model_format_pairs,
+    dtype: str,
+    max_tokens: int,
+    num_logprobs: int,
+) -> None:
+    model_name, sparsity = model_format_pairs
+
+    sparse_model = vllm_runner_sparse(model_name=model_name,
+                                      sparsity=sparsity,
+                                      dtype=dtype,
+                                      max_model_len=1024)
+    sparse_outputs = sparse_model.generate_greedy_logprobs(
+        example_prompts, max_tokens, num_logprobs)
+
+    # Note: not sure why, but deleting just the model on Ada Lovelace
+    #   does not free the GPU memory. On Ampere, deleting the just model
+    #   frees the memory.
+    del sparse_model.model.llm_engine.driver_worker
+    del sparse_model
+    torch.cuda.empty_cache()
+    gc.collect()
+
+    dense_model = vllm_runner_sparse(model_name,
```

Comment:
could you just use `vllm_runner` here to be very clear?

Reply:
I made a new class called `vllm_runner_sparse` which allows for passing the `sparsity` argument

This inherits from `vllm_runner`

Figured it would be easier to maintain relative to upstream if we had a separate class rather than editing the definition of `vllm_runner`

Reply:
i was just saying to use the `vllm_runner` for the dense model here, rather than using `vllm_runner_sparse` with `sparsity=None`

Reply:
Downside of that is that I need to pass `max_model_len`

I'm going to update the class name to be `vllm_runner_nm` rather than `vllm_runner_sparse`